clis/daemon-common/src/daemon.rs

```diff
@@ -69,33 +71,36 @@ impl Daemon {
     ///
     /// Current process type indicating whether it's the API or the Daemon.
     /// This is useful because of how the fork syscall.
-    pub fn start() -> Result<ProcessType> {
-        #[cfg(unix)]
-        {
-            use std::fs::File;
-            // This crate only works on unix systems and not Windows.
-            use daemonize::Daemonize;
-
-            let tclid_wd_path = get_wd_path()?;
-
-            std::fs::create_dir_all(&tclid_wd_path)?;
-            let stderr = File::create(tclid_wd_path.join("tclid.err"))?;
-
-            let daemonize = Daemonize::new()
-                .pid_file("tclid.pid") // Every method except `new` and `start`
-                .chown_pid_file(true) // is optional, see `Daemonize` documentation
-                .working_directory(tclid_wd_path) // for default behavior.
-                .stderr(stderr); // Redirect stderr to `/tmp/daemon.err`.
-
-            match daemonize.execute() {
-                daemonize::Outcome::Parent(Ok(daemonize::Parent { .. })) => Ok(ProcessType::Api),
-                daemonize::Outcome::Parent(Err(err)) => Err(err.into()),
-                daemonize::Outcome::Child(Ok(_)) => Ok(ProcessType::Daemon),
-                daemonize::Outcome::Child(Err(err)) => Err(err.into()),
-            }
+    fn start() -> Result<ProcessType> {
+        use std::fs::File;
+        let wd_path = get_wd_path(Self::NAME)?;
+
+        std::fs::create_dir_all(&wd_path)?;
+        let stderr = File::create(wd_path.join(format!("{}.err", Self::NAME)))?;
+
+        // Every method except `new` and `start` is optional, see `Daemonize` documentation for
+        // default behavior.
+        let daemonize = Daemonize::new()
```

Comment:
Turns out, this daemonization should be conditional (e.g. on config or cli argument) because according to OpenWRT docs, their [init system](https://openwrt.org/docs/guide-developer/procd-init-scripts)  expects daemons to be running in foreground and the procd itself takes care of daemonization.

This means that we need ability to start a daemon in "daemon" mode without daemonization (forking).

Reply:
Okk, I guess we can move all of this daemonization stuff under some feature? I guess we might need it for different OSes? Or we won't, so maybe I should remove it in the end? ^^'

Reply:
IMHO, we can leave the start/stop commands, for standard `tclid` usage, like it was before. But add something (e.g. probably a command or argument, like `teliod --daemon` or similar, whatever makes the most sense to you) that allows to start daemon and skip the forking part with the assumption, that forking is done by procd/systemd/etc.

Reply:
Hmm, but `tclid` or `teliod`?

Reply:
I mean, that `tclid` would have commands to start stop daemon, and `teliod` does not really need them.